### PR TITLE
Update ariakit to version 0.3.5

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Update `@ariakit/react` from version `0.3.3` to the (currently latest) version `0.3.5`

## Why?

To benefit from the latest additions and bug fixes (eg. several fixes to `Menu`  and its subcomponents)

## How?

Using npm.

From the [release notes](https://github.com/ariakit/ariakit/blob/main/packages/ariakit-react/CHANGELOG.md), there shouldn't be any breaking changes, although some changes in version `0.3.4` caused a few `Tabs` unit tests to fail — likely related to the change "Controlled store updates are no longer flushed synchronously."

I was able to fix the unit tests by:
- using `useLayoutEffect` instead of `useEffect` (see [this article](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect) explaining the differences)
- tweaking unit tests to be a bit more detailed in what to wait for

I also changed the `Tooltip` component to use the newly introduced `unmountOnHide`, instead of the previous custom logic.

## Testing Instructions

From the [release notes](https://github.com/ariakit/ariakit/blob/main/packages/ariakit-react/CHANGELOG.md), there shouldn't be any breaking changes, although some changes in version `0.3.4` caused a few `Tabs` unit tests to fail — likely related to the change "Controlled store updates are no longer flushed synchronously."

We should make sure that components using ariakit are still working as expected in Storybook and the editor:

- `TabPanel`
- `Tabs`
- `ToggleGroupControl`
- `Toolbar`
- `Tooltip`

### CODE REVIEW

File: packages/components/src/tabs/test/index.tsx
Comment (ciampo) [inline packages/components/src/tabs/test/index.tsx:1]: @chad1008 I realised that the `Tabs` unit tests still assume that `Tabs.Tab` accepts an `icon` prop — I guess we should remove that from the prop types and from the unit tests (in a separate PR) ?

**Review by brookewp — APPROVED** [review]
The mentioned components are testing well in both the editor and in SB 🎉 Nice to see the new prop in Tooltip!

**Review by diegohaz — APPROVED** [review]

**Review by chad1008 — APPROVED** [review]
LGTM, thanks for spotting those `icon` tests in `Tabs`! I'll get a PR together.

**Review by tyxla — APPROVED** [review]
Looks good to me 👍

File: packages/components/src/tooltip/index.tsx [hunk 1]
```diff
@@ -80,8 +78,9 @@ function Tooltip( props: TooltipProps ) {
 			>
 				{ isOnlyChild ? undefined : children }
 			</Ariakit.TooltipAnchor>
-			{ isOnlyChild && ( text || shortcut ) && isTooltipOpen && (
+			{ isOnlyChild && ( text || shortcut ) && (
 				<Ariakit.Tooltip
+					unmountOnHide
```
Comment (ntsekouras) [inline packages/components/src/tooltip/index.tsx:83]: I see some errors related to this `unmountOnHide`. (React does not recognize the `unmountOnHide` prop on a DOM element.). 

--edit

Sorry, disregard this. I hadn't npm installed 😅

Comment (tyxla) [reply packages/components/src/tooltip/index.tsx:83]: 😌 My heart jumped for a second. I just had to revert something in another repo 😅

Comment (vcanales) [reply packages/components/src/tooltip/index.tsx:83]: I'm seeing errors that seem related to @ntsekouras' comment:

```
packages/components/src/tooltip/index.tsx:82:5 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Props<{ as: As<any>; } & Omit<TooltipOptions<"div">, "as">>): ReactElement<any, string | JSXElementConstructor<any>> | null', gave the following error.
    Property 'as' is missing in type '{ children: (string | Element | undefined)[]; unmountOnHide: true; className: string; gutter: number; id: string | undefined; overflowPadding: number; store: TooltipStore; }' but required in type '{ as: As<any>; }'.
  Overload 2 of 2, '(props: Props<TooltipOptions<"div">>): ReactElement<any, string | JSXElementConstructor<any>> | null', gave the following error.
    Type '{ children: (string | Element | undefined)[]; unmountOnHide: true; className: string; gutter: number; id: string | undefined; overflowPadding: number; store: TooltipStore; }' is not assignable to type 'IntrinsicAttributes & TooltipOptions<"div"> & { [index: `data-${string}`]: unknown; wrapElement?: WrapElement | undefined; children?: Children<any>; render?: ReactElement<...> | ... 1 more ... | undefined; } & Omit<...>'.
      Property 'unmountOnHide' does not exist on type 'IntrinsicAttributes & TooltipOptions<"div"> & { [index: `data-${string}`]: unknown; wrapElement?: WrapElement | undefined; children?: Children<any>; render?: ReactElement<...> | ... 1 more ... | undefined; } & Omit<...>'.

 82     <Ariakit.Tooltip
        ~~~~~~~~~~~~~~~~
 83      unmountOnHide
    ~~~~~~~~~~~~~~~~~~
... 
 88      store={ tooltipStore }
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 89     >
    ~~~~~

  packages/components/node_modules/@ariakit/react-core/cjs/utils/types.d.ts:64:9
    64         as: T;
               ~~
    'as' is declared here.

packages/components/src/dropdown-menu-v2-ariakit/index.tsx:307:5 - error TS2322: Type '{ children: Element; modal: boolean; store: MenuStore<MenuStoreValues>; gutter: number; shift: number; hideOnHoverOutside: false; "data-side": string; ... 257 more ...; onTransitionEndCapture?: TransitionEventHandler<...> | undefined; }' is not assignable to type 'IntrinsicAttributes & MenuOptions<"div"> & { [index: `data-${string}`]: unknown; wrapElement?: WrapElement | undefined; children?: Children<any>; render?: ReactElement<...> | ... 1 more ... | undefined; } & Omit<...> & { ...; } & Pick<...>'.
  Property 'unmountOnHide' does not exist on type 'IntrinsicAttributes & MenuOptions<"div"> & { [index: `data-${string}`]: unknown; wrapElement?: WrapElement | undefined; children?: Children<any>; render?: ReactElement<...> | ... 1 more ... | undefined; } & Omit<...> & { ...; } & Pick<...>'.

307     unmountOnHide
        ~~~~~~~~~~~~~


Found 2 errors.
```

This is after clearing node_modules and re-installing. Using node@16.19.1 and npm@8.19.3 — No idea what I can do about this, really; any tips? I've been working with the previous version of Gutenberg meanwhile, which works fine.

cc. @ciampo

**Pull request merged**